server/hotrod/src/main/java/org/infinispan/server/hotrod/Encoder2x.java

```diff
@@ -475,14 +495,35 @@ private void writeHeaderNoTopology(ByteBuf buffer, long messageId, HotRodOperati
    }
 
    private void writeStatus(HotRodHeader header, ByteBuf buf, HotRodServer server, boolean compatibilityEnabled, OperationStatus status) {
-      if (server == null || HotRodVersion.HOTROD_24.isOlder(header.version))
+      if (server == null || HotRodVersion.HOTROD_24.isOlder(header.version) || HotRodVersion.HOTROD_29.isAtLeast(header.version))
          buf.writeByte(status.getCode());
       else {
-         OperationStatus st = OperationStatus.withCompatibility(status, compatibilityEnabled);
+         OperationStatus st = OperationStatus.withLegacyStorageHint(status, compatibilityEnabled);
```

Comment:
```withLegacyStorageHint``` is a strange name. Why do we avoid 'compatibility' from naming? I'd call this ```withLegacyCompatibility``` so we'll know from a glance ... :)

Reply:
The idea is to wipe compat mode, as long as variable names still refers to it, it'll not be truly dead!  :)

Reply:
I can put it back if you think it's essential, but regardless, I think storageHint reflects more precisely the fact that how data is stored in the cache piggybacks on the response of every operation

Reply:
So there will be no mention of it, not even in a migration guide? That would be so sad:(

Reply:
It will be in the migration guide for sure